lib/src/component_declaration/component_base.dart

```diff
@@ -514,28 +494,10 @@ abstract class UiProps extends MapBase
     }
   }
 
-  // Manually implement members from `MapViewMixin`,
-  // since mixing that class in doesn't play well with the DDC.
-  // TODO find out root cause and reduced test case.
-  @override operator[](Object key) => _map[key];
-  @override void operator[]=(key, value) { _map[key] = value; }
-  @override void addAll(other) { _map.addAll(other); }
-  @override void clear() { _map.clear(); }
-  @override putIfAbsent(key, ifAbsent()) => _map.putIfAbsent(key, ifAbsent);
-  @override bool containsKey(Object key) => _map.containsKey(key);
-  @override bool containsValue(Object value) => _map.containsValue(value);
-  @override void forEach(void action(key, value)) { _map.forEach(action); }
-  @override bool get isEmpty => _map.isEmpty;
-  @override bool get isNotEmpty => _map.isNotEmpty;
-  @override int get length => _map.length;
-  @override Iterable get keys => _map.keys;
-  @override remove(Object key) => _map.remove(key);
-  @override Iterable get values => _map.values;
-
   // Manually implement members from `StateMapViewMixin`,
```

Comment:
(I think `StateMapViewMixin` is a typo and should say `PropsMapViewMixin`.)

Could we try mixing in `PropsMapViewMixin instead and see how it works with Dart 2's DDC?